packages/plugin-infra/src/manager/sns-adaptor.ts

```diff
@@ -16,6 +17,16 @@ export function useActivatedPluginsSNSAdaptor() {
     return useSubscription(subscription)
 }
 
+export function useActivatedPluginSNSAdaptorWithChainRequirementMet() {
+    const chainId = useChainId()
+    const plugins = useActivatedPluginsSNSAdaptor()
+    return plugins.reduce((acc: { [key: string]: boolean }, cur) => {
```

Comment:
Why not use filter?

Reply:
I just need a `{ [plugin.ID]: boolean }` mappings to indicate does each plugin support the current chain.